src/preprocess_info.cc

```diff
@@ -158,7 +195,162 @@ Napi::Value PreProcessInfo::GetMeanVariant(const Napi::CallbackInfo& info) {
     return Napi::Object::New(env);
   }
 
-  return Napi::String::New(
-      env, utils::GetNameOfMeanVariant(_input_info->getPreProcess().getMeanVariant()));
+  return Napi::String::New(env,
+                           utils::GetNameOfMeanVariant(
+                               _input_info->getPreProcess().getMeanVariant()));
+}
+
+Napi::Value PreProcessInfo::GetPreProcessChannel(
+    const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+
+  if (info.Length() != 1) {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+        .ThrowAsJavaScriptException();
+    return Napi::Object::New(env);
+    ;
+  }
+
+  if (!info[0].IsNumber()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return Napi::Object::New(env);
+    ;
+  }
+
+  size_t index = info[0].ToNumber().Int32Value();
+  ie::PreProcessInfo& preInfo = _input_info->getPreProcess();
+
+  size_t numberOfChannels = preInfo.getNumberOfChannels();
+
+  if (numberOfChannels == 0) {
+    Napi::Error::New(env, "accessing pre-process when nothing was set.")
+        .ThrowAsJavaScriptException();
+    return env.Null();
+  };
+
+  if (index >= numberOfChannels) {
+    std::string errorLog =
+        "pre process index " + std::to_string(index) + " is out of bounds.";
+    Napi::Error::New(env, errorLog).ThrowAsJavaScriptException();
+    return env.Null();
+  };
+
+  const ie::PreProcessChannel::Ptr& preProcessChannel = preInfo[index];
+  Napi::Object preprocess_channel = Napi::Object::New(env);
+  preprocess_channel.Set("stdScale", preProcessChannel->stdScale);
+  preprocess_channel.Set("meanValue", preProcessChannel->meanValue);
+
+  ie::Blob::Ptr meandata = preProcessChannel->meanData;
+  std::unique_ptr<ie::LockedMemory<void>> locked_memory_;
+  ie::MemoryBlob::Ptr memory_meandata = ie::as<ie::MemoryBlob>(meandata);
+  if (!memory_meandata) {
+    preprocess_channel.Set("meanData",
+                           Napi::ArrayBuffer::ArrayBuffer(env, env.Null()));
+    Napi::Value channelInfo = Napi::Value::From(env, preprocess_channel);
+
+    return channelInfo;
+  }
+
+  locked_memory_.reset(reinterpret_cast<ie::LockedMemory<void>*>(
+      new ie::LockedMemory<const void>(memory_meandata->rmap())));
+  Napi::ArrayBuffer buffer = Napi::ArrayBuffer::New(
```

Comment:
The lifecycle of `locked_memory_` is different from the newly created ArrayBuffer. So you need to create a new ArrayBuffer with pre-allocated memory by `Napi::ArrayBuffer::New (napi_env env, size_t byteLength)` and copy the data of `memory_meandata` to it. BTW, because you copy the data, so it doesn't require to hold the memory, just use a local variable `local_memory` would be fine.

Reply:
I keep this way.

Reply:
This way might cause the memory access violation error. Please change according to my suggestion. Thanks.

Reply:
@huningxin Hi, ningxin. I meet a block here. I have created a new ArrayBuffer with the bytelength only. But I can not find the vaild way to change the data in the ArrayBuffer variable. The API `Data( )` of the ArrayBuffer returns a pointer the wrapped data, but it seems `const` variable. Here is the description of the API:

```
void* Napi::ArrayBuffer::Data() const;

Returns a pointer the wrapped data.
```

Cloud you introduce some way to copy the data of `memory_meandata` to the ArrayBuffer variable ?

Reply:
I have done it by using `memcpy()`.